packages/core/lib/console.js

```diff
@@ -16,18 +16,25 @@ const Require = require("@truffle/require");
 const debug = require("debug")("console");
 const { getCommand } = require("./command-utils");
 
-const processInput = input => {
-  const inputComponents = input.trim().split(" ");
-  if (inputComponents.length === 0) return input;
+const processInput = (input, allowedCommands) => {
+  const words = input.trim().split(/\s+/);
+  if (words.length === 0) return input;
 
-  if (inputComponents[0] === "truffle") {
-    return inputComponents.slice(1).join(" ");
+  if (words[0] === "truffle") {
+    if (allowedCommands.includes(words[1])) {
+      return words.slice(1).join(" ");
+    }
+    // Log friendly information
+    // return console.info expression which, when processed, will not modify
+    // the `_` variable.
+    return `console.info("ℹ️ : 'truffle ${words[1]}' is not allowed within Truffle REPL")`;
```

Comment:
We also need to check for disallowed commands in the case the user enters input in the following form: `<command> --<option> <value>...`.

Besides that I think this is also the wrong place for this check, no? I feel like the input should be processed (removing the "truffle") and then the check should occur. Do you disagree?

Reply:
would moving the validation part to a different function make sense here?

Reply:
Yea, this will have to be refactored, to handle validation at this seam, and `Console`'s clients and tests will have to be updated as well. 🦆 🦆 🦆 :/

Reply:
Turns out that there wasn't a need to refactor as we can grab the list of valid truffle commands, use it to determine branching logic.